### PR TITLE
Grid movement

### PR DESCRIPTION
**Pull request by kegriz**
Grid movement and onclick movement based on "grid-engine" library

### CODE REVIEW

File: src/classes/player.ts [hunk 1]
```diff
@@ -10,8 +10,8 @@ export class Player extends Actor {
   private keyDown: Phaser.Input.Keyboard.Key;
   private keyRight: Phaser.Input.Keyboard.Key;
```
Comment (kelostrada) [inline src/classes/player.ts:11]: I think those key fields are unnecessary now

File: src/scenes/moon/index.ts [hunk 2]
```diff
@@ -39,13 +40,76 @@ export class Moon extends Scene {
...
+      } else {
+        npcs.get(charId).anims.play(direction);
+      }
+    });
+
+    this.gridEngine.movementStopped().subscribe(({ charId, direction }: any) => {
+      if (charId == "player") {
+        this.player.anims.stop();
+        this.player.anims.play("stay-down");
+      }
```
Comment (kelostrada) [inline src/scenes/moon/index.ts:98]: we can consider adding stop animations for other characters as well :) but this is for tests so i wouldnt bother, just something we can consider :)

File: src/classes/player.ts (hunk 1 above)
Comment (kegriz) [reply src/classes/player.ts]: done

File: src/scenes/moon/index.ts (hunk 2 above)
Comment (kegriz) [reply src/scenes/moon/index.ts]: done

**Pull request merged**